### ISSUE TITLE
BusyIndicator:  After showing a modal the indicator shows busy, moving the mouse to the top of form it switches off

### ISSUE THREAD

**Issue by petercullen68** [infor-design/enterprise-ng#582]
**Describe the bug**
I have a REST call which before it's called sets the busy indicator and then sets it off.  Sometimes the response to this call needs a modal with a yes/no button.  An answer to this question refires the call.  At this point many times the busy indicator remains running but the variable controlling it is set to false.  If I move the mouse to the banner at the top of the form it will reset and the indicator stops.  Note:  If I don't show the modal but fire the call over and over, this never happens.  It seems to be related to the modal being shown.

**To Reproduce**
Our project can be cloned and run to show the exact issue.  I can also provide the code to show where it is happening.

**Expected behavior**
I would expect the indicator to reflect the variable it's bound to.

**Version**
- ids-enterprise-ng: v5.5.1

**Screenshots**
![Start](https://user-images.githubusercontent.com/42790047/63049862-b6f77f80-bea7-11e9-9297-d9697680317b.png)
![part1](https://user-images.githubusercontent.com/42790047/63049988-0047cf00-bea8-11e9-9f4a-6fbd464de60d.png)
![part2](https://user-images.githubusercontent.com/42790047/63049989-0047cf00-bea8-11e9-8ac7-f3df783cb42e.png)
![part3](https://user-images.githubusercontent.com/42790047/63049990-0047cf00-bea8-11e9-9425-3b00828fd095.png)
![part4](https://user-images.githubusercontent.com/42790047/63049991-0047cf00-bea8-11e9-80b0-93f4bf6b766e.png)
[base.destroyer.ts.txt](https://github.com/infor-design/enterprise-ng/files/3502973/base.destroyer.ts.txt)
[quality.assurance.component.html.txt](https://github.com/infor-design/enterprise-ng/files/3502974/quality.assurance.component.html.txt)
[quality.assurance.component.ts.txt](https://github.com/infor-design/enterprise-ng/files/3502975/quality.assurance.component.ts.txt)

**Platform**
 - Device (if applicable) [e.g. iPhone 6 or Samsung Galaxy S6]
 - OS Version: [Windows 10]
 - Browser Name [chrome]
 - Browser Version [

(Official Build) (64-bit)]

**Additional context**

**Comment by tmcconechy** [comment]
Ideas:
- Would it be better if you just showed the busy indicator after you ask the question and close the form? 
- Alternatively you can show the busy indicator IN the modal http://master-enterprise.demo.design.infor.com/components/busyindicator/test-inside-modal.html
- but not sure about the part about hovering the header?

Is it possible to update one of the examples in this repo to show the problem in a reduced test case? This would greatly lower the time needed to invest in looking into and fixing the issue.

**Comment by petercullen68** [comment]
The flow of the program is this..
1) Select an item on grid.
2) Just before call made, show busy, directly after call response, set off.
3) Ask question.  Answer no.
4) Call made again, busy set on, after call response set off...
5) Busy indicator still spinning.  Move the mouse to the banner at the top, busy stops.

Also if place a breakpoint inside the logic this prevents the issue from happening as well.  It's a timing issue I believe.

I'll see if I can mock up something in a sample project.

**Comment by petercullen68** [comment]
[TWL RF - Google Chrome 2019-08-14 15-52-14.zip](https://github.com/infor-design/enterprise-ng/files/3503187/TWL.RF.-.Google.Chrome.2019-08-14.15-52-14.zip)

**Comment by petercullen68** [comment]
I have modified the form.demo.html and form.demo.ts (and added a new .ts to be placed in the busyindicator folder which shows this behavior.  I have attached the changed files which can be placed into the enterprise-ng-5.5.1 project.  I've also attached an MP4 showing the behavior.  I see that the issue has been changed to an enhancement.  I would think this is a bug, when I set the busy flag to false, the indicator should not stay spinning.  
[busyindicator.take.two.mp4.zip](https://github.com/infor-design/enterprise-ng/files/3507731/busyindicator.take.two.mp4.zip)
[busyindicator.zip](https://github.com/infor-design/enterprise-ng/files/3507732/busyindicator.zip)

**Comment by petercullen68** [comment]
Some code snippets..  

html:

 <button soho-button (click)="doDialog(true)">Fire wCall</button>

soho-busyindicator
      [activated]="this.busyIam"
      [transparentOverlay]="transparentOverlay"
      [blockUI]="true"
      [text]="message"
      [displayDelay]="delay"
      registerForEvents="afterstart complete"
      (afterstart)="onAfterStart($event)"
      (complete)="onComplete($event)"


.ts

  doDialog(showDialog: boolean) {
    this.start();
    this.http.get('https://www.infor.com').subscribe(
      data => {
        this.stop();
        if (showDialog) {
          this.showYesNo('1234', '4231');
          this.doDialog(false);
        }
        console.log('POST Request is successful ', data);
      },
      error => {
        this.stop();
        if (showDialog) {
          this.showYesNo('123', '231', null, (result: string) => {
          this.doDialog(false);
        });
        console.log('Error', error);
      }
    });
  }

  start() {
    this.busyIam = true;
    console.log('start');
  }

  stop() {
    this.busyIam = false;
    console.log('stop');
  }

The start and stop methods are being called and so the busy indicator should not keep spinning after doDialog has been called.

**Comment by tmcconechy** [comment]
@petercullen68 Im finding the videos really hard to follow? Do you think you would be able to add steps to reproduce on the issue that go along with the sample? I tink that would make this issue easier to look at. I.E. change the issue form the perspective of the new sample not from the way your app works so we can look at it from that standpoint?

Watching the video im not really sure whats wrong, seems like you press the buttons and the busy indicator shows.

Also maybe as a workaround you can play with the delay setting. I dont think its needed to always show the busy indicator if the thing your doing is fairly quick.

**Comment by petercullen68** [comment]
Hey Tim, basically if you press the button marked Fire wcall, the busy indicator continues to spin after all logic is completed.  The variable bound to the activated status has been set to false.  But the spinner keeps spinning.

If you press the stop button to stop it, it sets this variable to false, but it's already false when you press it.

**Comment by tmcconechy** [comment]
OK I see. Seems a bit complicated maybe to solve. What about just using `this.busyIndicator.activated = true;` and `this.busyIndicator.activated = false;` like the original example and not having a binding to the activated event. Since we dont have an example like this i didnt realize it was even possible.

**Comment by petercullen68** [comment]
Yep I can try that.  We have a slightly complicated use case, as we bind to two different properties, one for busy and one for network not being available.  I have a base class that I can code much of this.

**Comment by pwpatton** [comment]
Does forcing doDialog into the next tick help? Thinking this will allow current tick to complete the busy indicator activated binding before processing the dialog in the next tick. 
```
  doDialog(showDialog: boolean) {
    this.start();
    this.http.get('https://www.infor.com').subscribe(
      data => {
        this.stop();
        setTimeout(() => {
          if (showDialog) {
            this.showYesNo('1234', '4231');
            this.doDialog(false);
          }
        });
        console.log('POST Request is successful ', data);
      },
```

**Comment by petercullen68** [comment]
Thanks for the suggestion pwpatton.  I had actually refactored at this point as per Tim's suggestion, so I didn't get a chance to try what you suggested.

**Comment by pwpatton** [comment]
@petercullen68 no problem. I should have answered earlier.

**Comment by tmcconechy** [comment]
Just closing this for now as you have a solution and we dont really have the time right now for a better fix
